assets/src/edit-story/components/inspector/prepublish/components/checklistTab.js

```diff
@@ -243,81 +212,81 @@ const ChecklistTab = ({
+      currentCheckpoint === PPC_CHECKPOINT_STATE.UNAVAILABLE ||
+      isChecklistEmpty ||
+      (!isRecommendedVisible && !isHighPriorityVisible)
+    );
+  }, [
+    currentCheckpoint,
+    isRecommendedVisible,
+    isHighPriorityVisible,
+    isChecklistEmpty,
+  ]);
 
-  return isChecklistEmpty ? (
-    <EmptyChecklist />
-  ) : (
+  if (isEmptyView) {
+    return <EmptyChecklist body={TEXT.UNAVAILABLE_BODY} />;
+  }
+
+  return (
     <>
-      <SimplePanel
-        isToggleDisabled={isHighPriorityDisabled}
-        name="checklist"
-        hasBadge
-        title={
-          <>
-            <PanelTitle isDisabled={isHighPriorityDisabled}>
-              {TEXT.HIGH_PRIORITY_TITLE}
-            </PanelTitle>
-            {!isHighPriorityDisabled && (
+      {isHighPriorityVisible && (
+        <SimplePanel
+          name="checklist"
+          hasBadge
+          title={
+            <>
+              <PanelTitle>{TEXT.HIGH_PRIORITY_TITLE}</PanelTitle>
               <NumberBadge number={highPriorityLength} />
-            )}
-          </>
-        }
-        ariaLabel={TEXT.HIGH_PRIORITY_TITLE}
-      >
-        {!isHighPriorityDisabled && (
-          <Entries
-            items={highPriority}
-            pageItems={pages.highPriority}
-            renderRow={renderRow}
-            renderPageGroupedRow={renderPageGroupedRow}
-            disabled={isHighPriorityDisabled}
-          />
-        )}
-      </SimplePanel>
-      <SimplePanel
-        isToggleDisabled={isRecommendedDisabled}
-        name="checklist"
-        hasBadge
-        title={
-          <>
-            <PanelTitle isRecommended isDisabled={isRecommendedDisabled}>
-              {TEXT.RECOMMENDED_TITLE}
-            </PanelTitle>
-            {!isRecommendedDisabled && (
-              <NumberBadge isRecommended number={recommendedLength} />
-            )}
-          </>
-        }
-        ariaLabel={TEXT.RECOMMENDED_TITLE}
-      >
-        {canOptimizeVideo && (
-          <AutoVideoOptimization
-            areVideosAutoOptimized={areVideosAutoOptimized}
-            onAutoOptimizeVideoClick={onAutoVideoOptimizationClick}
-          />
-        )}
-        {!isRecommendedDisabled && (
-          <Entries
-            items={recommended}
-            pageItems={pages.recommended}
-            renderRow={renderRow}
-            renderPageGroupedRow={renderPageGroupedRow}
-            disabled={isRecommendedDisabled}
-          />
-        )}
-      </SimplePanel>
+            </>
+          }
+          ariaLabel={TEXT.HIGH_PRIORITY_TITLE}
+        >
+          {highPriority.map(renderRow)}
+          {Object.entries(pages.highPriority || {}).map(renderPageGroupedRow)}
+        </SimplePanel>
+      )}
+      {isRecommendedVisible && (
+        <SimplePanel
+          name="checklist"
+          hasBadge
+          title={
+            <>
+              <PanelTitle isRecommended>{TEXT.RECOMMENDED_TITLE}</PanelTitle>
+              {recommendedLength > 0 && (
```

Comment:
this check is for cases where recommended _is_ empty but there's still an option to optimize video on upload. We don't want to show `0` as a badge.